src/init.ts

```diff
@@ -102,3 +57,99 @@ export async function checkCLI(cli: string): Promise<boolean> {
   log.debug(`Exit code for running ${cli} is ${code}`);
   return code === 0;
 }
+
+async function getInstalledClis(): Promise<Config> {
+  const config: Config = emptyConfig;
+
+  for (const platform in MeshPlatform) {
+    const mp = platform as MeshPlatform;
+    if (await checkCLI(PlatformCommand[mp])) {
+      config.connected[mp] = true;
+    }
+    log.debug(`CLI ${PlatformCommand[mp]} installed.`);
+  }
+
+  return config;
+}
+
+async function checkIfConfigExists() {
+  const exist = await exists(configFilePath);
+  if (!exist) {
+    log.info(
+      "No configuration file found. I will create one for you :)",
+    );
+    log.info(
+      "... searching for installed cloud CLIs",
+    );
+    const config = await getInstalledClis();
+    const platforms: string[] = [];
+    Object.keys(config.connected).forEach((val) => {
+      const con = val as MeshPlatform;
+      if (config.connected[con]) {
+        platforms.push(val);
+      }
+    });
+    const answer = await askYesNo(
+      `I see you already have these cloud CLIs installed: ${
+        platforms.join(", ")
+      }.\nDo you want to connect these to ${CLIName}?`,
+    );
+    if (answer) {
+      await writeConfig(config);
+      log.info(
+        `Saved your configuration here: ${configFilePath}. Have fun using ${CLIName}!`,
+      );
+    } else {
+      log.info(
+        `Ok, we did not connect the cloud CLIs. Run "${CLICommand} config -h" to see how to individually connect a new cloud CLI to ${CLIName}.`,
+      );
+    }
+  }
+}
+
+export async function init() {
+  // check if config file exists
+  if (!(Deno.args[0] === "config")) {
+    await checkIfConfigExists();
+
+    // check if config file is valid
+    const config = parseJsonWithLog<Config>(readFile(configFilePath));
+
+    if (!objectsHaveSameKeys(emptyConfig, config)) {
+      log.error(
+        `Configuration does not match the required format. Please delete the config File at ${configFilePath} and run "${CLICommand} config" again`,
```

Comment:
Are these duplicate messages necessary?

Reply:
true does not make sense as we now log the error on the top level :D